httpe/httpe.go

```diff
@@ -0,0 +1,51 @@
+// Package httpe holds http server utilities
+//nolint: gochecknoglobals
+package httpe
+
+import (
+	"net/http"
+)
+
+type HandlerE interface {
+	ServeHTTP(http.ResponseWriter, *http.Request) error
+}
+
+// HandlerFuncE works like an HTTP.HandlerFunc with the addition of an error
+// return value. It is intended to be used with WriteErr which handles
+// the error and turns it into an appropriate http response StatusCode and
+// Body.
+type HandlerFuncE func(http.ResponseWriter, *http.Request) error
+
+// ServeHTTP calls f(w, r) and returns its error.
+func (f HandlerFuncE) ServeHTTP(w http.ResponseWriter, r *http.Request) error {
+	return f(w, r)
+}
+
+// WriteErr translates an error into the appropriate http response
+// StatusCode and Body and writes it.
+type WriteErr func(http.ResponseWriter, error)
```

Comment:
I keep coming on this and thinking it should be an interface:
```
type ErrorWriter interface {
    Write(http.ResponseWriter, error)
}

type ErrorWriterFunc func(http.ResponseWriter, error)
func (ew ErrorWriterFunc) Write(w http.ResponseWriter, err error) {
    ew(w, err)
}
```

I feel like I'm going overboard though. But when I see that `WriteErr` type as a function, I keep thinking it should be an interface.

Reply:
This might actually be quite useful.
Logging or so for instance could be part of CustomerHandlerE state that also implments ErrorWriter.... Why not. Let's try.

Reply:
I went with `httpe.ErrWriterFunc` (long enough).

Reply:
Done.